### PR TITLE
Oceanwater 605 antennae faults in rqt

### PR DESCRIPTION
**Pull request by lanseafood**
## Linked Issues:
| Jira Ticket 🎟️   | [Oceanwater-605](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-605) |
| ----------- | ----------- |
| EPIC ⚡| [Oceanwater-551](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-551) |
| Github :octocat:  | na |

## Summary of Changes
* New pant and tilt faults in Rqt
* on select, populate JPL messages for pan/tilt message in /pt_faults_status

## Test
* open any roslaunch
* go to rqt, dynamic recongfigure, and select from either a pan or tilt fault
* rosoptic echo /pt_faults_status and see the value update to 1 (hardware fault)

### CODE REVIEW

**Review by kmdalal — APPROVED** [review]
I just looked at the code.  Don't have time to test today.  Suggest adding another reviewer if you'd like another tester.

File: ow_faults/include/ow_faults/FaultInjector.h [hunk 1]
```diff
@@ -39,14 +40,17 @@ class FaultInjector
   enum ComponentFaults {
```
Comment (Samahu) [inline ow_faults/include/ow_faults/FaultInjector.h:40]: **Effective Modern C++: Item 10 - Prefer using scoped enums to unscoped enums**
```suggestion
  enum class ComponentFaults : int {
```

File: ow_faults/include/ow_faults/FaultInjector.h [hunk 2]
```diff
@@ -39,14 +40,17 @@ class FaultInjector
...
-    TrajectoryGeneration=2, 
+    TrajectoryGeneration = 2,
     Collision=3, 
     Estop=4, 
     PositionLimit=5, 
     TorqueLimit=6, 
     VelocityLimit=7, 
-    NoForceData=8};
+    NoForceData=8
+    };
```
Comment (Samahu) [inline ow_faults/include/ow_faults/FaultInjector.h:53]: Can you stay consistent in the spaces between the enum and the value? I think C++ community prefers adding a singe space before and after the equal sign.

File: ow_faults/src/FaultInjector.cpp [hunk 3]
```diff
@@ -211,12 +221,22 @@ void FaultInjector::jointStateCb(const sensor_msgs::JointStateConstPtr& msg)
...
 
+  if (m_faults.antennae_pan_failure){
+    systemFaultsBitmask |= isPanTiltExecutionError;
+    setPTFaultsMessage(pt_faults_msg,hardwareFault);
+  }
+  if (m_faults.antennae_tilt_failure) {
+    systemFaultsBitmask |= isPanTiltExecutionError;
+    setPTFaultsMessage(pt_faults_msg,hardwareFault);
+  }
+
```
Comment (Samahu) [inline ow_faults/src/FaultInjector.cpp:232]: Refactoring suggestion to reduce the code duplication
```suggestion
  if (m_faults.antennae_pan_failure || m_faults.antennae_tilt_failure) {
    systemFaultsBitmask |= isPanTiltExecutionError;
    setPTFaultsMessage(pt_faults_msg,hardwareFault);
  }
```

File: ow_faults/include/ow_faults/FaultInjector.h [hunk 4]
```diff
@@ -97,6 +102,7 @@ class FaultInjector
   ros::Publisher m_fault_status_pub;
   ros::Publisher m_arm_fault_status_pub;
   ros::Publisher m_power_fault_status_pub;
+  ros::Publisher m_antennae_fault_status_pub;
```
Comment (Samahu) [inline ow_faults/include/ow_faults/FaultInjector.h:105]: Just trying to understand the design better, why do we need to have a faults publisher per fault category?

**Review by Samahu — COMMENTED** [review]
Worked as described, minor code improvements desired

File: ow_faults/include/ow_faults/FaultInjector.h (hunk 4 above)
Comment (kmdalal) [reply ow_faults/include/ow_faults/FaultInjector.h:105]: I suggest adding Terry as reviewer, as he wrote the original.  Perhaps some design decisions won't scale up well.  I'm not sure about this one.

File: ow_faults/include/ow_faults/FaultInjector.h (hunk 2 above)
Comment (lanseafood) [reply ow_faults/include/ow_faults/FaultInjector.h:53]: will do

File: ow_faults/include/ow_faults/FaultInjector.h (hunk 4 above)
Comment (lanseafood) [reply ow_faults/include/ow_faults/FaultInjector.h:105]: will add Terry

File: ow_faults/msg/PTFaults.msg [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#Pan/Tilt fault status
+Header header
+int32 value
```
[the remark this answers is not in the document]
Comment (lanseafood) [reply ow_faults/msg/PTFaults.msg]: will do

File: ow_faults/cfg/Faults.cfg [hunk 6]
```diff
@@ -10,13 +10,14 @@ joint_state_enum = gen.enum([gen.const("nominal",  int_t, 0, "Joint is functioni
                              gen.const("frozen",   int_t, 2, "Joint is frozen in position"),
                              gen.const("friction", int_t, 3, "Joint is consuming extra power")],
                             "An enum to set joint state")
-# ARM FAULTS
+# ANTENNAE FAULTS
```
Comment (Samahu) [inline ow_faults/cfg/Faults.cfg:13]: I think it is antenna
```suggestion
# ANTENNA FAULTS
```

File: ow_faults/src/FaultInjector.cpp [hunk 7]
```diff
@@ -85,118 +98,122 @@ void FaultInjector::jointStateCb(const sensor_msgs::JointStateConstPtr& msg)
   }
 
   sensor_msgs::JointState output(*msg);
+  std_msgs::Float64 soc_msg;
+
```
Comment (Samahu) [inline ow_faults/src/FaultInjector.cpp:102]: why did you need to move this up?

**Review by Samahu — APPROVED** [review]

File: ow_faults/src/FaultInjector.cpp (hunk 7 above)
Comment (lanseafood) [reply ow_faults/src/FaultInjector.cpp:102]: just so that it would live with all the other declarations. It felt more random in the middle.

Comment (Samahu) [reply ow_faults/src/FaultInjector.cpp:102]: Well in C++ it is best if you delay the declaration of stack variables till the point you are going to use it.
In any case, this really hints that this method is quite long .. consider breaking it into smaller parts later on (i.e when you start another related-PR)
